Update to Stencil 0.7

kylef opened the pull request:
Closes #217 

I noticed you're using a beta version of CocoaPods that I'm not, please feel free to pull down my PR and run `pod install` on a different version if you'd rather that didn't change.

kylef commented:
This should be good to go now, looks like Travis CI has a large backlog.

djbe commented:
@kylef I'll also do a pod install like you suggested, to minimise the diff (after your rebase).

kylef commented:
I've updated this PR with the following:

- Rebased onto current master.
- Switched to same version of CocoaPods as master.
- Added changelog item.
- Updated SwiftGen template documentation to refer to new Stencil documentation website.

AliSoftware commented on CHANGELOG.md:
```diff
@@ -19,6 +19,9 @@
 * Strings dot-syntax template: use `enum`s for namespacing instead of `struct`s.  
   [David Jennes](https://github.com/djbe)
   [#218](https://github.com/AliSoftware/SwiftGen/pull/218)
+* Updated Stencil to [0.7](https://github.com/kylef/Stencil/releases/tag/0.7.0).  
```
Seems you actually updated to 0.7.2 :wink:

kylef replied:
Yes, although 0.7 includes 0.7.0, 0.7.1, 0.7.2 😉.

I think it's better to link to 0.7 changelog because 0.7.1 and 0.7.2 do not include changes that will affect any user of SwiftGen. These two releases only include bug fixes for problems in 0.7.

plivesey commented:
Nice @kylef. Commenting so I get an email when it's merged.
